src/vit-servicing-station/vit-servicing-station-tests/src/tests/rest/search.rs

```diff
@@ -0,0 +1,483 @@
+use crate::common::clients::SearchRequestBuilder;
+use crate::common::startup::quick_start;
+use assert_fs::TempDir;
+use vit_servicing_station_lib::db::models::challenges::Challenge;
+use vit_servicing_station_lib::db::models::proposals::FullProposalInfo;
+use vit_servicing_station_lib::v0::endpoints::search::requests::{
+    Column, SearchQuery, SearchResponse,
+};
+
+#[test]
+pub fn search_challenges_by_title() {
+    let temp_dir = TempDir::new().unwrap();
+    let (server, data) = quick_start(&temp_dir).unwrap();
+    let rest_client = server.rest_client_with_token(&data.token_hash());
+    let expected_challenge = &data.challenges()[0];
+
+    let response = rest_client
+        .search(
+            SearchRequestBuilder::default()
+                .on_challenges()
+                .by_title(&expected_challenge.title)
+                .into(),
+        )
+        .unwrap();
+
+    assert_response_contains_challenge(expected_challenge, response);
+}
+
+fn assert_response_contains_challenge(
+    expected_challenge: &Challenge,
+    search_response: SearchResponse,
+) {
+    if let SearchResponse::Challenge(challenges) = search_response {
```

Comment:
Tiny nit - we can save a bit of indentation and keep the panic close with `let SearchReponse::Challenge(challenges) = search_response else { panic!() }